packages/amplify_datastore/lib/amplify_datastore.dart

```diff
@@ -326,3 +312,42 @@ class _NativeAmplifyApi
   @override
   String get runtimeTypeName => '_NativeAmplifyApi';
 }
+
+/// {@template amplify_datastore.datastore_plugin_options}
+/// The plugin options for the Amplify DataStore plugin.
+/// {@endtemplate}
+class DataStorePluginOptions {
```

Comment:
I think the plugin option definitions should live in core

Reply:
I think we should maintain category-specific types within the core and keep types exclusive to plugins within their respective package.